### PR TITLE
Fix permissions on /dev/kvm

### PR DESCRIPTION
**Pull request by keszybz**
This fixes the issue I was complaining about on telegram.

### CODE REVIEW

File: src/udev/udev-event.c [hunk 1]
```diff
@@ -834,39 +833,28 @@ static int update_devnode(UdevEvent *event) {
...
-                                /* default 0660 if a group is assigned */
-                                event->mode = 0660;
-                        else
-                                /* default 0600 */
-                                event->mode = 0600;
-                }
         }
+        if (event->mode == MODE_INVALID && gid_is_valid(event->gid) && event->gid > 0)
+                /* If group is set, but mode is not set, "upgrade" mode for the group. */
+                event->mode = 0660;
```
Comment (yuwata) [inline src/udev/udev-event.c:853]: Why we should not "upgrade" mode when `gid == 0`?

**Comment by yuwata** [comment]
The change looks nice, but only one question, and I guess it is critical for why my previous code introduces the issue.

File: src/udev/udev-event.c (hunk 1 above)
Comment (keszybz) [reply src/udev/udev-event.c:853]: > Why we should not "upgrade" mode when gid == 0?

The kernel default is sufficient in this case (mode 0600). 

>  why my previous code introduces the issue.

Your code introduced the issue by adding `if (r == -ENOENT) uid=0;` and similar for gid and mode.

Comment (yuwata) [reply src/udev/udev-event.c:853]: > > Why we should not "upgrade" mode when gid == 0?
> 
> The kernel default is sufficient in this case (mode 0600).

What happens on 'change' uevent?

> > why my previous code introduces the issue.
> 
> Your code introduced the issue by adding `if (r == -ENOENT) uid=0;` and similar for gid and mode.

Yeah, I understand that point. Thanks.

Comment (yuwata) [reply src/udev/udev-event.c:853]: > > > Why we should not "upgrade" mode when gid == 0?
> > 
> > 
> > The kernel default is sufficient in this case (mode 0600).
> 
> What happens on 'change' uevent?

Ah, in that case, `mode != MODE_INVALID`, isn't it?

**Pull request merged**